Tools/environment_install/installROS-ubuntu.sh

```diff
@@ -0,0 +1,234 @@
+
+# Setup sources.lst
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+# Setup keys
+sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+# If you experience issues connecting to the keyserver, you can try substituting hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 in the previous command.
+# Installation
+
+heading "${green}Updating apt${reset}"
+sudo apt update
+
+heading "${green}Installing ROS${reset}"
+# Here we loop through any packages passed on the command line
+# Install packages ...
+for package in "${packages[@]}"; do
+  sudo apt install $package -y
+done
+
+# This is where you might start to modify the packages being installed, i.e.
+# sudo apt install ros-${ROS_VERSION}-{package_name}
+# sudo apt install -y setpriv
+# sudo apt install -y ros-${ROS_VERSION}-robot-upstart
+# sudo apt install -y ros-${ROS_VERSION}-navigation
+
+# install other needed packages
+sudo apt install build-essential cmake libglfw3-dev libglew-dev libeigen3-dev libjsoncpp-dev libtclap-dev -y
+
+#
+# To find available packages:
+# apt-cache search ros-melodic
+# 
+# Initialize rosdep
+
+heading "${green}Installing rosdep${reset}"
+
+sudo apt install python3-rosdep -y
+# Certificates are messed up on earlier version Jetson for some reason
+# Do not know if it is an issue with the Xavier, test by commenting out
+# sudo c_rehash /etc/ssl/certs
+# Initialize rosdep
+
+heading "${green}Initializaing rosdep${reset}"
+
+
+sudo rm /etc/ros/rosdep/sources.list.d/20-default.list && echo "OK" || echo "No 20-default.list file"
+
+sudo rosdep init
+# To find available packages, use:
+rosdep update
+# Use this to install dependencies of packages in a workspace
+# rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
+# Environment Setup - Don't add /opt/ros/${ROS_VERSION}/setup.bash if it's already in bashrc
+heading "${green}Adding setup.bash, ROS_MASTER_URI and ROS_HOSTNAME to .bashrc ${reset}"
+grep -q -F "ROS_HOSTNAME=$HOSTNAME.local" ~/.bashrc || echo "ROS_HOSTNAME=$HOSTNAME.local" >> ~/.bashrc
+grep -q -F "ROS_MASTER_URI=http://$HOSTNAME.local:11311" ~/.bashrc || echo "ROS_MASTER_URI=http://$HOSTNAME.local:11311" >> ~/.bashrc
+grep -q -F "source /opt/ros/${ROS_VERSION}/setup.bash" ~/.bashrc || echo "source /opt/ros/${ROS_VERSION}/setup.bash" >> ~/.bashrc
+source ~/.bashrc
+
+
+heading "${green}Installing rosinstall tools${reset}"
+
+sudo apt install python3-rosinstall python3-rosinstall-generator python3-wstool build-essential python3-catkin-tools -y
+
+heading "${green}Installing Ardupilot-ROS workspace${reset}"
+
+if maybe_prompt_user "Add ardupilot-ws to your home folder [N/y]?" ; then
+    if [ ! -d $ROS_WS_ROOT ]; then
+        mkdir -p $ROS_WS_ROOT/src
+        pushd $ROS_WS_ROOT
+        catkin init
+        pushd src
+        git clone https://github.com/snktshrma/ardupilot_ros.git
```

Comment:
this package is not catkin compliant ... the rosdep  will install nothing and we got not info on what it serves for

Reply:
Yes, this is still a work on progress being developed under GSoC